### PR TITLE
[SPARK-41134][SQL] Improve error message of internal errors

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

Spark is an open system where users can use plugins in different places. When people hit a bug, it may not come from Spark, but from these plugins. We should make this clear in the error message.

### Why are the changes needed?

better error message

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @MaxGekk @zsxwing

File: sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala [hunk 1]
```diff
@@ -494,7 +494,8 @@ object QueryExecution {
   private[sql] def toInternalError(msg: String, e: Throwable): Throwable = e match {
     case e @ (_: java.lang.NullPointerException | _: java.lang.AssertionError) =>
       SparkException.internalError(
-        msg + " Please, fill a bug report in, and provide the full stack trace.",
+        msg + " You hit a bug in Spark or the Spark plugins you use. Please, report this bug " +
```
[the remark this answers is not in the document]
Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala:497]: Spark should be responsible for its dependencies, as it's not the users but Spark that decides which library to depend on. Spark plugins are users' choice.

[the remark this answers is not in the document]
Comment (MaxGekk) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala:497]: Makes sense.

**Review by MaxGekk — APPROVED** [review]

File: sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala (hunk 1 above)
Comment (xkrogen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala:497]: Somewhat unrelated to the point of Spark vs. plugins, but at least for Spark, it seems we should provide guidance here on _how_ a user should report a bug.

Comment (cloud-fan) [reply sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala:497]: it's hard to give an answer here... stackoverflow, dev list, JIRA ticket, slack channels, etc.

**Comment by cloud-fan** [comment]
thanks for review, merging to master!

**Pull request closed without merging**